code/graphics/opengl/gropenglshader.cpp

```diff
@@ -296,12 +316,11 @@ void opengl_shader_set_current(int handle)
 
 size_t opengl_get_shader_idx(shader_type shader_t, unsigned int flags) 
 {
-	for (size_t idx = 0; idx < GL_shader.size(); idx++) {
-		if (GL_shader[idx].shader == shader_t && GL_shader[idx].flags == flags) {
-			return idx;
-		}
+	auto t = shader_descriptor_t(shader_t, flags);
```

Comment:
You could avoid a copy operation here and improve the variable name. `t` isn't very descriptive...

```suggestion
	shader_descriptor_t descriptor(shader_t, flags);
```

Reply:
Thanks for the review/feedback.

I will do this tonight.

Also, I tested locally, and everything looks fine. I'm afraid that the performance gain may be from lost shaders though that I didn't notice. I'm not saying my code is bad, but this is the most in-depth patch I've done so far, so please excercise caution while reviewing. :)

Reply:
I would have told you if I found something wrong with the changes :stuck_out_tongue:

I looked though the code and your changes seem to make sense. I didn't compile and run the code though...

Reply:
So... you can't have a bad variable name or a copy if you don't have a variable. ;)